data_source/gateway_test.go

```diff
@@ -101,3 +115,139 @@ func TestAdaptStateUpdate(t *testing.T) {
 		}
 	}
 }
+
+func TestAdaptClass(t *testing.T) {
+	response := new(clients.ClassDefinition)
+	err := json.Unmarshal(classJson, response)
+	assert.NoError(t, err)
+
+	class, err := adaptClass(response)
+	assert.NoError(t, err)
+
+	assert.Equal(t, new(felt.Felt).SetUint64(0), class.APIVersion)
+
+	for i, v := range response.EntryPoints.External {
+		assert.Equal(t, v.Selector, class.Externals[i].Selector)
+		assert.Equal(t, v.Offset, class.Externals[i].Offset)
+	}
+	assert.Equal(t, len(response.EntryPoints.External), len(class.Externals))
+
+	for i, v := range response.EntryPoints.L1Handler {
+		assert.Equal(t, v.Selector, class.L1Handlers[i].Selector)
+		assert.Equal(t, v.Offset, class.L1Handlers[i].Offset)
+	}
+	assert.Equal(t, len(response.EntryPoints.L1Handler), len(class.L1Handlers))
+
+	for i, v := range response.EntryPoints.Constructor {
+		assert.Equal(t, v.Selector, class.Constructors[i].Selector)
+		assert.Equal(t, v.Offset, class.Constructors[i].Offset)
+	}
+	assert.Equal(t, len(response.EntryPoints.Constructor), len(class.Constructors))
+
+	for i, v := range response.Program.Builtins {
+		assert.Equal(t, new(felt.Felt).SetBytes([]byte(v)), class.Builtins[i])
+	}
+	assert.Equal(t, len(response.Program.Builtins), len(class.Builtins))
+
+	for i, v := range response.Program.Data {
+		assert.Equal(t, new(felt.Felt).SetBytes([]byte(v)), class.Bytecode[i])
+	}
+	assert.Equal(t, len(response.Program.Data), len(class.Bytecode))
+
+	programHash, err := clients.ProgramHash(response)
+	assert.NoError(t, err)
+	assert.Equal(t, programHash, class.ProgramHash)
+}
+
+func TestAdaptInvokeTransaction(t *testing.T) {
+	response := new(clients.TransactionStatus)
+	err := json.Unmarshal(invokeJson, response)
+	assert.NoError(t, err)
+
+	invokeTx := adaptInvokeTransaction(response)
+	assert.Equal(t, response.Transaction.ContractAddress, invokeTx.ContractAddress)
+	assert.Equal(t, response.Transaction.EntryPointSelector, invokeTx.EntryPointSelector)
+	assert.Equal(t, response.Transaction.SenderAddress, invokeTx.SenderAddress)
+	assert.Equal(t, response.Transaction.Nonce, invokeTx.Nonce)
+	assert.Equal(t, response.Transaction.Calldata, invokeTx.CallData)
+	assert.Equal(t, response.Transaction.Signature, invokeTx.Signature)
+	assert.Equal(t, response.Transaction.MaxFee, invokeTx.MaxFee)
+	assert.Equal(t, response.Transaction.Version, invokeTx.Version)
+}
+
+func getMockClass() (*clients.ClassDefinition, *core.Class) {
+	classJson, _ := os.ReadFile("testdata/class.json")
+
+	response := new(clients.ClassDefinition)
+	json.Unmarshal(classJson, response)
+	class, _ := adaptClass(response)
+
+	return response, class
+}
```

Comment:
I am a little confused by this function. Is this still required if you are using go embed?

Reply:
oh you are right. i totally forgot to remove os.ReadFile. 
we stil need this function because we want `client.ClassDefinition` and `core.Class`